pkg/portal/graphql/viewer.go

```diff
@@ -48,6 +53,19 @@ var nodeViewer = node(
 			return nil, nil
 		}
 
-		return gqlCtx.Users.Load(id).Value, nil
+		userIface, err := gqlCtx.Users.Load(id).Value()
+		if err != nil {
+			return nil, err
+		}
+
+		user := userIface.(*model.User)
+
+		requestIP := httputil.GetIP(gqlCtx.Request, false)
```

Comment:
trustProxy should be injected, instead of using a hard-coded value of `false`. You can inject TrustProxy into Context.

```diff
diff --git a/pkg/portal/graphql/context.go b/pkg/portal/graphql/context.go
index 50ecf3cc2..e15347d2c 100644
--- a/pkg/portal/graphql/context.go
+++ b/pkg/portal/graphql/context.go
@@ -166,6 +166,7 @@ type Context struct {
        Request   *http.Request
        GQLLogger Logger

+       TrustProxy              config.TrustProxy
        Users                   UserLoader
        Apps                    AppLoader
        Domains                 DomainLoader
diff --git a/pkg/portal/wire_gen.go b/pkg/portal/wire_gen.go
index 9baefa4be..6c372e9fb 100644
--- a/pkg/portal/wire_gen.go
+++ b/pkg/portal/wire_gen.go
@@ -89,6 +89,7 @@ func newGraphQLHandler(p *deps.RequestProvider) http.Handler {
        request := p.Request
        logFactory := rootProvider.LoggerFactory
        logger := graphql.NewLogger(logFactory)
+       trustProxy := environmentConfig.TrustProxy
        authgearConfig := rootProvider.AuthgearConfig
        adminAPIConfig := rootProvider.AdminAPIConfig
        controller := rootProvider.ConfigSourceController
@@ -169,7 +170,6 @@ func newGraphQLHandler(p *deps.RequestProvider) http.Handler {
        inProcessQueue := &task.InProcessQueue{
                Executor: inProcessExecutor,
        }
-       trustProxy := environmentConfig.TrustProxy
        httpHost := deps.ProvideHTTPHost(request, trustProxy)
        httpProto := deps.ProvideHTTPProto(request, trustProxy)
        requestOriginProvider := &endpoint.RequestOriginProvider{
@@ -341,6 +341,7 @@ func newGraphQLHandler(p *deps.RequestProvider) http.Handler {
        graphqlContext := &graphql.Context{
                Request:                 request,
                GQLLogger:               logger,
+               TrustProxy:              trustProxy,
                Users:                   userLoader,
                Apps:                    appLoader,
                Domains:                 domainLoader,
```